sql/src/main/java/io/crate/planner/consumer/ManyTableConsumer.java

```diff
@@ -511,9 +511,7 @@ private static void visitFieldsRequiredLater(java.util.function.Consumer<Field>
                 // it's not necessary to extend the output
                 continue;
             }
-            if (relations.contains(leftName) || relations.contains(rightName)) {
-                FieldsVisitor.visitFields(symbol, fieldConsumer);
-            }
+            FieldsVisitor.visitFields(symbol, fieldConsumer);
```

Comment:
I thought this was important for semi-joins - would it be possible to run the tests once with semi-join rewrite always enabled - just to be sure?

Reply:
Just ran all tests with `UseSemiJoins` set to `1.0`. Seems to work fine.